tests/unit/components/sl-menu-item-test.js

```diff
@@ -0,0 +1,141 @@
+        this.$().hasClass( 'active' ),
+        false,
+        'Rendered element does not have class "active" by default'
+    );
+
+    Ember.run( () => {
+        component.set( 'active', true );
+    });
+
+    assert.ok(
+        this.$().hasClass( 'active' ),
+        'Rendered element has class "active" when component is active'
+    );
+});
+
+test( 'Class "has-sub-menu" is present when bound item has items array', function( assert ) {
+    const component = this.subject();
+
+    assert.ok(
+        false === this.$().hasClass( 'has-sub-menu' ),
+        'Rendered element does not have class "has-sub-menu" by default'
+    );
+
+    Ember.run( () => {
+        component.set( 'item', { items: [ { label: '' } ] } );
+    });
+
+    assert.ok(
+        this.$().hasClass( 'has-sub-menu' ),
+        'Rendered element has class "has-sub-menu"'
+    );
+});
+
+test( 'Class "active" is present when bound item is selected', function( assert ) {
+    const component = this.subject();
+
+    assert.ok(
+        false === this.$().hasClass( 'active' ),
+        'Rendered element does not have class "active" by default'
+    );
+
+    Ember.run( () => {
+        component.set( 'item', { selected: true } );
+    });
+
+    assert.ok(
+        this.$().hasClass( 'active' ),
+        'Rendered element has class "active"'
+    );
+});
+
+test( 'Component responds to mouse enter/leave events', function( assert ) {
+    const component = this.subject();
+
+    assert.ok(
+        false === component.get( 'active' ),
+        'Component is not active by default'
+    );
+
+    assert.ok(
+        false === this.$().hasClass( 'active' ),
+        'Rendered element does not have "active" class'
+    );
+
+    Ember.run( () => {
+        this.$().trigger( 'mouseenter' );
+    });
+
+    assert.ok(
+        true === component.get( 'active' ),
+        'Component is active after mouseenter event'
+    );
+
+    assert.ok(
+        this.$().hasClass( 'active' ),
+        'Rendered element has "active" class after mouseenter event'
+    );
+
+    Ember.run( () => {
+        this.$().trigger( 'mouseleave' );
+    });
+
+    assert.ok(
+        false === component.get( 'active' ),
+        'Component is inactive after mouseleave event'
+    );
+
+    assert.ok(
+        false === this.$().hasClass( 'active' ),
+        'Rendered element is inactive after mouseleave event'
+    );
+});
+
+test( 'Bound item.action is triggered when link is clicked', function( assert ) {
+    this.subject({
+        action: 'test',
+
+        item: {
+            action: 'test',
+            data: { okay: true },
+            label: 'Test'
+        },
+
+        targetObject: {
+            test( actionName, data ) {
+                assert.ok(
+                    'test' === actionName && true === data.okay,
+                    'Bound item.action triggered with expected data'
+                );
+            }
+        }
+    });
+
+    this.$( 'a' ).trigger( 'click' );
+});
```

Comment:
I'd prefer to see `hasSubItems` tested specifically, and individually.  If we remove the class bindings against its output then there is no longer any test coverage against this method.  Granted, at the moment, nothing else is using the value but that's not really the point.  The point is, we are testing the functionality of this method by testing how its output is used by something else, and not by testing the method directly.